server/routes/induction/update/affectAbilityToWorkUpdateController.ts

```diff
@@ -0,0 +1,82 @@
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import createError from 'http-errors'
+import type { InductionDto } from 'inductionDto'
+import type { AffectAbilityToWorkForm } from 'inductionForms'
+import AffectAbilityToWorkController from '../common/affectAbilityToWorkController'
+import toCreateOrUpdateInductionDto from '../../../data/mappers/createOrUpdateInductionDtoMapper'
+import logger from '../../../../logger'
+import { InductionService } from '../../../services'
+import validateAffectAbilityToWorkForm from './affectAbilityToWorkFormValidator'
+
+/**
+ * Controller for the Update of the Factors Affecting a Prisoner's Ability To Work screen of the Induction.
+ */
+export default class AffectAbilityToWorkUpdateController extends AffectAbilityToWorkController {
+  constructor(private readonly inductionService: InductionService) {
+    super()
+  }
+
+  getBackLinkUrl(req: Request): string {
+    const { prisonNumber } = req.params
+    return `/plan/${prisonNumber}/view/work-and-interests`
+  }
+
+  getBackLinkAriaText(_req: Request): string {
+    return 'Back to <TODO - check what CIAG UI does here>'
+  }
+
+  submitAffectAbilityToWorkForm: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ): Promise<void> => {
+    const { prisonNumber } = req.params
+    const { prisonerSummary, inductionDto } = req.session
+    const { prisonId } = prisonerSummary
+
+    req.session.affectAbilityToWorkForm = { ...req.body }
+    if (!req.session.affectAbilityToWorkForm.affectAbilityToWork) {
+      req.session.affectAbilityToWorkForm.affectAbilityToWork = []
+    }
```

Comment:
If you prefer, you could use a Nullish coalescing assignment here (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment).

```suggestion
  req.session.affectAbilityToWorkForm.affectAbilityToWork ??= []
```

Reply:
That would be more succinct, but I'd like to solicit opinion on that 👍

Reply:
I think if we do it, then yes, Matt is right, we should do it in the other controllers we are writing at the moment (they all follow this pattern).
But also we should also do it in conjunction with the next line, as they basically work together. As it stands the code is this:

```
            if (!req.session.affectAbilityToWorkForm.affectAbilityToWork) {
	      req.session.affectAbilityToWorkForm.affectAbilityToWork = []
	    }
            if (!Array.isArray(req.session.affectAbilityToWorkForm.affectAbilityToWork)) {
              req.session.affectAbilityToWorkForm.affectAbilityToWork = [
                 req.session.affectAbilityToWorkForm.affectAbilityToWork,
              ]
            }
```
The purpose of these 2 if statements is to make sure the form object in the session is "correct" before we start. Specifically we are looking to "correct" up the array field - `affectAbilityToWork` in this case.

This is to make our validators simpler (otherwise they would have to do this type of thing), but getting the array in order beforehand will also help our mapper code later on.

The problem is to do with how the browser submits the form with checkbox fields, and how it is deserialized into the form object.
The html form binds the checkboxes to the input field name `affectAbilityToWork`
If the user selects and the browser submits more than one checkbox value, then the form submission contains many `affectAbilityToWork` fields which are deserialized into an array - happy days
If the user does not select any checkboxes at all, the browser does not submit the field at all, and hence the property is null. This is the first `if` statement.
If the user only selects one checkbox, the browser submits it .... but only a single `affectAbilityToWork` field, which gets deserialized into a string property, not an array property. This is what the 2nd if statement is fixing.

So, more than happy to do something with the current pattern (though I would argue that as it currently stands its more readable than other approaches?), but we should do it in conjunction with the 2nd if statement as well. Perhaps refactor it out to a private function, or better still, refactor it out into a single  generic function somewhere so that it works with any form type and property and we can use it from all the controllers?

Reply:
I wasn't sure about the readability either, but that might be because of my lack of familiarity in this area and obviously it's always good to challenge that. Regardless, I'll leave it as it is for now and we can take a view on it separately.